test/e2e/framework/util.go

```diff
@@ -1249,15 +1249,16 @@ func (f *Framework) MatchContainerOutput(
 
 	if podErr != nil {
 		// Pod failed. Dump all logs from all containers to see what's wrong
-		for _, container := range podStatus.Spec.Containers {
-			logs, err := e2epod.GetPodLogs(f.ClientSet, ns, podStatus.Name, container.Name)
+		_ = podutil.VisitContainers(&podStatus.Spec, func(c *v1.Container) bool {
```

Comment:
`VisitContainers` searches initContainer also as https://github.com/kubernetes/kubernetes/blob/master/pkg/api/pod/util.go#L37
so lgtm for making debugging more easy.